clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```diff
@@ -298,7 +298,7 @@ private void sendUnsentRequests(final Timer timer) {
         do {
```

Comment:
Do we need the early return?
If there is a request in in-flight but unSentRequestQueue is empty, we will lose the newest request.